pkgs/by-name/ka/kakoune-lsp/package.nix

```
@@ -1,34 +1,37 @@
-{ stdenv, lib, fetchFromGitHub, rustPlatform, perl, CoreServices, Security, SystemConfiguration }:
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, substituteAll, perl
+, stdenv, CoreServices, Security, SystemConfiguration
+}:
 
 rustPlatform.buildRustPackage rec {
   pname = "kakoune-lsp";
-  version = "16.0.0";
+  version = "17.0.1";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-d4Tc6iYp20uOKMd+T2LhWgXWZzvzq1E+VWqjhhiIiHE=";
+    sha256 = "sha256-uXKquAjfytUn/Q0kx+0BGRQTkVMQ9rMRnTCy622upag=";
   };
 
-  cargoHash = "sha256-kV8d0PwIWS6gyfCtv70iv8MrL91ZOZbwYznhc3lUw0U=";
-
-  buildInputs = [ perl ] ++ lib.optionals stdenv.isDarwin [ CoreServices Security SystemConfiguration ];
-
   patches = [
-    ./Use-full-Perl-path.patch
+    (substituteAll {
+      src = ./Hardcode-perl.patch;
+      perl = lib.getExe perl;
+    })
```

- https://github.com/NixOS/nixpkgs/issues/237216
- https://github.com/NixOS/nixpkgs/pull/301350

```suggestion
    (replaceVars ./Hardcode-perl.patch { inherit perl; })
```